api/src/main/java/org/openmrs/api/ConditionService.java

```diff
@@ -73,6 +74,16 @@
 	@Authorized({ PrivilegeConstants.GET_CONDITIONS })
 	List<Condition> getAllConditions(Patient patient) throws APIException;
 
+	/**
+	 * Gets all conditions(not voided) of an encounter.
+	 * 
+	 * @param encounter - the encounter to retrieve conditions for
+	 * @return a list of encounter's conditions
+	 * @throws APIException
+	 * @since 2.4.0
+	 */
+	List<Condition> getConditionsByEncounter(Encounter encounter) throws APIException;
```

Comment:
@dkayiwa / @ibacher what do you think, should this just be `getConditionsForEncounter(Encounter)` or even `getConditions(Encounter)`?

Reply:
I like the explicitness of `getConditionsForEncounter()`.

Reply:
The undocumented convention is `getConditionsByEncounter` 😊